lib/view_component/base.rb

```diff
@@ -117,11 +128,12 @@ def render_in(view_context, &block)
             end
           end.to_s
 
-        # Avoid allocating new string when output_preamble and output_postamble are blank
-        if output_preamble.blank? && output_postamble.blank?
-          rendered_template
+        if cache_key.present?
+          Rails.cache.fetch(@vc_cache_key) do
```

Comment:
```suggestion
          Rails.cache.fetch(cache_key) do
```

Lets avoid relying on this ivar since it requires `cache_key` to be called first. We should access one or the other consistently.